std/math.d

```diff
@@ -240,8 +240,350 @@ enum RealFormat
     ieeeQuadruple,
 }
 
-// Constants used for extracting the components of the representation.
-// They supplement the built-in floating point properties.
+/* Low-level access to the individual parts of a floating-point value.
+
+   TODO: Make this publically available through std.bitmanip once the API
+   is stable and tested.
+*/
+union RealRep(N)
+    if (isFloatingPoint!N &&  is(N == Unqual!N))
+{
+pure: @safe: nothrow: @nogc:
+pragma(inline, true):
+
+    static if (N.mant_dig == 11 && N.max_exp == 0x6)
+    {
+        enum format = RealFormat.ieeeHalf,
+             excessDig = 0,
+             storeWhole = false,
+             expDig = 5;
+        alias Mant = ushort,
+              Exp = byte;
```

Comment:
@ibuclaw I have now documented `RealRep` rather thoroughly. Please take another look at this PR.